components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt

```diff
@@ -141,33 +164,16 @@ internal class ConstellationObserver(
 /**
  * An [AutoPushFeature] observer to handle [FxaAccountManager] subscriptions and push events.
  */
-internal class PushObserver(
-    private val accountManager: FxaAccountManager
-) : Bus.Observer<PushType, String>, PushSubscriptionObserver {
-
-    private val logger = Logger("PushObserver")
-
-    override fun onSubscriptionAvailable(subscription: AutoPushSubscription) {
-        logger.debug("Received new push subscription from $subscription.type")
-
-        if (subscription.type == PushType.Services) {
-            accountManager.withConstellation {
-                it.setDevicePushSubscriptionAsync(
-                    DevicePushSubscription(
-                        endpoint = subscription.endpoint,
-                        publicKey = subscription.publicKey,
-                        authKey = subscription.authKey
-                    )
-                )
-            }
-        }
-    }
+internal class AutoPushObserver(private val accountManager: FxaAccountManager) : AutoPushFeature.Observer {
+    private val logger = Logger(AutoPushObserver::class.java.simpleName)
 
-    override fun onEvent(type: PushType, message: String) {
-        logger.debug("Received new push message for $type")
+    override fun onMessageReceived(scope: String, message: ByteArray?) {
+        logger.info("Received new push message for $scope")
```

Comment:
We don't need to from our previous conversations since fxa will just drop messages it can't parse, but I agree, a simple check here since we have the context of our own scope short-circuits that out a lot quicker.

Reply:
The less possibility for things to go wrong, the better :)